pymongo/mongo_client.py

```diff
@@ -2150,3 +2186,35 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         return self.handle(exc_type, exc_val)
+
+
+def _before_fork():
+    # Ensure that we aren't in any critical region.
+    _acquire_locks()
```

Comment:
Don't we have to acquire the _clients_lock here and release it after-fork (like we do for _insertion_lock)?

Reply:
Ping.